nengo/networks/assoc_mem.py

```diff
@@ -330,46 +481,109 @@ def add_wta_network(self, inhibit_scale=1.5, inhibit_synapse=0.005):
+
+            utility = getattr(self, output_utilities_name)
+
+            nengo.Connection(self.bias_node, self.bias_ens1.input,
                              transform=np.ones((self.n_items, 1)),
                              synapse=None)
-            nengo.Connection(self.bias_node, self.thresh_ens.input,
+            nengo.Connection(self.bias_node, self.bias_ens2.input,
                              transform=np.ones((self.n_items, 1)),
                              synapse=None)
-            nengo.Connection(self.elem_output, self.thresh_bias.input,
-                             transform=-inhibit_scale)
-            nengo.Connection(self.thresh_bias.output, self.thresh_ens.input,
+            nengo.Connection(utility, self.bias_ens1.input,
                              transform=-inhibit_scale)
+            nengo.Connection(self.bias_ens1.output, self.bias_ens2.input,
+                             transform=-1.0)
+
+            # --- Make the output node and connect it
+            output_vectors = self._output_vectors[output_name]
+            cleanup_output_node = nengo.Node(size_in=output_vectors.shape[1],
+                                             label=cleanup_output_name)
+            nengo.Connection(self.bias_ens2.output, cleanup_output_node,
+                             transform=output_vectors.T, synapse=None)
 
-            # Reroute connections from elem_output to be connections
-            # from thresh_ens.output
-            def reroute(conn):
-                c = nengo.Connection(self.thresh_ens.output, conn.post,
-                                     transform=conn.transform,
-                                     synapse=conn.synapse)
-                self.connections.remove(conn)
-                return c
-            self.default_vector_inhibit_conns = [
-                reroute(c) for c in self.default_vector_inhibit_conns]
-            self.out_conns = [reroute(c) for c in self.out_conns]
-
-            # Make inhibitory connection if inhibit option is set
+            setattr(self, cleanup_output_name, cleanup_output_node)
+            setattr(self, cleanup_utilities_name, self.bias_ens2.output)
+
+            # --- Replace the original output node (pointer) if required
+            if replace_output:
+                setattr(self, output_name, cleanup_output_node)
+
+            # --- Make inhibitory connection if inhibit option is set
             if self.inhibit is not None:
-                for e in self.thresh_ens.ensembles:
+                for e in self.bias_ens2.ensembles:
                     nengo.Connection(
                         self.inhibit, e, transform=-self._inhib_scale,
                         synapse=None)
+
+            # --- Connect default output vector to cleaned outputs
+            #     (if available)
+            default_vector_ens_name = '_'.join([output_name,
+                                                self.default_ens_suffix])
```

Comment:
I would do `vector_ens = getattr(self, '_'.join([output_name, self.defaut_ens_suffix]), None)` and then on the next line do `if vector_ens`. That saves you having to store the name and use `hasattr`.